habitat-lab/habitat/tasks/rearrange/actions/oracle_nav_action.py

```diff
@@ -267,3 +267,299 @@ def step(self, *args, is_last_action, **kwargs):
+        """
+        The function checks if the agent collides with the object
+        given the navmesh
+        """
+        nav_pos_3d = [
+            np.array([xz[0], 0.0, xz[1]]) for xz in self._config.navmesh_offset
+        ]
+        cur_pos = [trans.transform_point(xyz) for xyz in nav_pos_3d]
+        cur_pos = [
+            np.array([xz[0], self._sim.articulated_agent.base_pos[1], xz[2]])
+            for xz in cur_pos
+        ]
+
+        for pos in cur_pos:  # noqa: SIM110
+            # Return true if the pathfinder says it is not navigable
+            if not self._sim.pathfinder.is_navigable(pos):
+                return True
+
+        return False
+
+    def rotation_collision_check(
+        self,
+        next_pos,
+    ):
+        """
+        This function checks if the robot needs to do backing-up action
+        """
+        # Make a copy of agent trans
+        trans = mn.Matrix4(self._sim.articulated_agent.sim_obj.transformation)
+        # Initialize the velocity controller
+        vc = SimpleVelocityControlEnv(self._config.sim_freq)
+        angle = float("inf")
+        # Get the current location of the agent
+        cur_pos = self._sim.articulated_agent.base_pos
+        # Set the trans to be agent location
+        trans.translation = self._sim.articulated_agent.base_pos
+
+        while abs(angle) > self._config.turn_thresh:
+            # Compute the robot facing orientation
+            rel_pos = (next_pos - cur_pos)[[0, 2]]
+            forward = np.array([1.0, 0, 0])
+            robot_forward = np.array(trans.transform_vector(forward))
+            robot_forward = robot_forward[[0, 2]]
+            angle = get_angle(robot_forward, rel_pos)
+            vel = OracleNavAction._compute_turn(
+                rel_pos, self._config.turn_velocity, robot_forward
+            )
+            trans = vc.act(trans, vel)
+            cur_pos = trans.translation
+
+            if self.is_collision(trans):
+                return True
+
+        return False
+
+    def step(self, *args, is_last_action, **kwargs):
+        nav_to_target_idx = kwargs[
+            self._action_arg_prefix + "oracle_nav_with_backing_up_action"
+        ]
+        if nav_to_target_idx <= 0 or nav_to_target_idx > len(
+            self._poss_entities
+        ):
+            if is_last_action:
+                return self._sim.step(HabitatSimActions.base_velocity)
+            else:
+                return {}
+        nav_to_target_idx = int(nav_to_target_idx[0]) - 1
+
+        final_nav_targ, obj_targ_pos = self._get_target_for_idx(
+            nav_to_target_idx
+        )
+        # Get the base transformation
+        base_T = self.cur_articulated_agent.base_transformation
+        # Get the current path
+        curr_path_points = self._path_to_point(final_nav_targ)
+        # Get the robot position
+        robot_pos = np.array(self.cur_articulated_agent.base_pos)
+
+        if curr_path_points is None:
+            raise Exception
```

Comment:
This seems like a bug. You probably want to raise an instationed instance not a class. And probably should be a RuntimeError